### ISSUE TITLE
Specify value of parameter is a random generated value

### ISSUE THREAD

**Issue by markcowl** [Azure/typespec-azure#759]
transferred from: https://github.com/microsoft/typespec/issues/3234

### Clear and concise description of the problem

In cases like `x-ms-client-request-id`, `datetime.now()`, we want to be able to express that the value for a header should be a random value generated, and users shouldn't have to input this.

```tsp
op foo(@header `x-ms-client-request-id`: $guid): void;
```

Also open to this being a tcgc decorator, but it feels best if we can incorporate this into the language

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://typespec.io/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by markcowl** [comment]
(Comment from @timotheeguerin)

is this set by the client or the server?

for server we can imagine have that

```
op test(): {
  @header requestId: utcDateTime = utcDateTime.now();
}
```

**Comment by iscai-msft** [comment]
I don't think this should be a tcgc change, but more of a compiler change. I'm going to remove the tcgc label from this, if you think differently @markcowl can you reassign it to tcgc? I just don't think we can add actual syntax here

**Comment by markcowl** [comment]
Some questions:
- Is this limited to GUID/uuid values? If so, can we just define a constructor for GUID?
- Do we need to specify who is responsible for generating the value (note that a request value would be generated by the client and a response value would be generated by the service
